### ISSUE TITLE
Support entering simple arithmetic expressions in DragValue

### ISSUE THREAD

**Issue by xTibor** [emilk/egui#1733]
There's an useful feature in a couple open-source graphics apps (Blender, Inkscape, Krita, etc.) where you can enter simple arithmetic expressions in numerical input boxes. This feature would also be nice to have in egui's DragValue.

I know introducing random dependencies is a no-go, however just as a test I slapped [the first expression evaluator crate](https://crates.io/crates/meval) I found on crates.io onto [DragValue](https://github.com/emilk/egui/blob/7eeb292adfacd9311a420ac3ea225e2261a8f8d3/egui/src/widgets/drag_value.rs#L200) to show how this could look like:

https://user-images.githubusercontent.com/1627292/173275380-96de369b-7007-4adf-be50-b72604807acf.mp4

**Comment by blackguest** [comment]
I like it +1

**Comment by emilk** [comment]
I like this too (I used to have this in Algodoo back in the day).

Definitely behind a feature flag though!

**Comment by emilk** [comment]
I like this too (I used to have this in Algodoo back in the day).

Definitely behind a feature flag though, and fins the lightest possible dependency!

**Comment by notgull** [comment]
I like this! Although I'd maybe have a configuration option regarding whether or not to enable this, not only at compile time through features but as an attribute on the widget too

**Comment by xTibor** [comment]
Fixed by #1967. You can now plug your favorite expression evaluator library into `.custom_parser()` and have 100% the same behavior as in my video above.

```rust
ui.add(
    DragValue::new(&mut self.my_value)
        .custom_parser(|s| meval::eval_str(s).ok())
);
```
